packages/common/src/types/field.ts

```diff
@@ -66,6 +66,7 @@ export interface Dimension extends Field {
 
 export interface CompiledDimension extends Dimension {
     compiledSql: string; // sql string with resolved template variables
+    tablesReferences: Array<string>;
```

Comment:
The compiled dimension/metric now indicates all the tables it has references to.